src/main/java/com/nukkitx/nbt/NbtMap.java

```diff
@@ -362,6 +362,11 @@ public boolean equals(Object o) {
         if (m.size() != size())
             return false;
 
+        if (this.hashCodeGenerated && o instanceof NbtMap && ((NbtMap) o).hashCodeGenerated &&
+                this.hashCode != ((NbtMap) o).hashCode) {
```

Comment:
Wouldn't it be possible to return the hashCode check instead of a dedicated false?

Reply:
We can not just rely on only comparing hashCodes since there can be two different objects with same hashCode. That's why it returns false instead of `return o1.hashCode() == o2.hashCode()`

Reply:
ah you're right, I somehow thought it was always unique 😅